server/methods/core/orders.js

```diff
@@ -62,53 +62,115 @@ export function ordersInventoryAdjust(orderId) {
  */
 export const methods = {
   /**
+   * orders/shipmentPicked
+   *
+   * @summary update picking status
+   * @param {Object} order - order object
+   * @param {Object} shipment - shipment object
+   * @return {Object} return workflow result
+   */
+  "orders/shipmentPicked": function (order, shipment) {
+    check(order, Object);
+    check(shipment, Object);
+
+    if (!Reaction.hasPermission("orders")) {
+      throw new Meteor.Error("access-denied", "Access Denied");
+    }
+
+    // Set the status of the items as picked
+    const itemIds = shipment.items.map((item) => {
+      return item._id;
+    });
+
+    const result = Meteor.call("workflow/pushItemWorkflow", "coreOrderItemWorkflow/picked", order, itemIds);
+    if (result === 1) {
+      return Orders.update({
+        "_id": order._id,
+        "shipping._id": shipment._id
+      }, {
+        $set: {
+          "shipping.$.workflow.status": "coreOrderWorkflow/picked"
+        }, $push: {
+          "shipping.$.workflow.workflow": "coreOrderWorkflow/picked"
+        }
+      });
+    }
+    return result;
+  },
+  /**
    * orders/shipmentPacked
    *
    * @summary update packing status
    * @param {Object} order - order object
    * @param {Object} shipment - shipment object
-   * @param {Boolean} packed - packed status
    * @return {Object} return workflow result
    */
-  "orders/shipmentPacked": function (order, shipment, packed) {
+  "orders/shipmentPacked": function (order, shipment) {
```

Comment:
See comments on unifying status progression methods in the shipmentPicked comment.